app/src/main/java/com/eventyay/organizer/core/auth/start/StartFragment.java

```diff
@@ -46,24 +49,40 @@
     private StartFragmentBinding binding;
     private Validator validator;
     private SharedViewModel sharedViewModel;
+    private SharedPreferences sharedPreferences;
+    private SharedPreferences.Editor editor;
+    private String warningPreferenceKey = "experimental_warning_shown";
+
 
     public static StartFragment newInstance() {
         return new StartFragment();
     }
 
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         binding = DataBindingUtil.inflate(inflater, R.layout.start_fragment, container, false);
         startFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(StartViewModel.class);
         sharedViewModel = ViewModelProviders.of(getActivity()).get(SharedViewModel.class);
         validator = new Validator(binding);
+        sharedPreferences = requireActivity().getSharedPreferences(warningPreferenceKey, Context.MODE_PRIVATE);
+        editor = sharedPreferences.edit();
```

Comment:
Write this logic in ViewModel

Reply:
Sir having the logic in viewModel. We have to deal with Shared Preferences as a live data which I think is of no use.
Thank you

Reply:
Okay Sir

Reply:
sir should handle the AlertDialog in a method or it is okay to call them in the lambda itself

Reply:
lambda